src/js/components/DateInput/utils.js

```diff
@@ -203,3 +199,37 @@ export const valuesAreEqual = (value1, value2) =>
     Array.isArray(value2) &&
     value1.every((d1, i) => d1 === value2[i])) ||
   value1 === value2;
+
+export const getTimestamp = (date) =>
+  new RegExp(/T.*/).test(date)
+    ? new Date(date).toISOString().split('T')[1]
+    : // for Calendar, explicitly mark that caller has provided
+      // value with no timestamp
+      false;
+
+// Adjust for differences between timestamp on value and
+// local timezone of user. Internal Calendar logic relies
+// on Javascript date contructor which translates the provided
+// date into the equivalent moment for the user's timezone, which
+// can create undesired results. The standardizes the input value
+// for internal calculations
+// Reference: https://www.ursahealth.com/new-insights/dates-and-timezones-in-javascript
+export const getAdjustedDate = (value, timestamp) => {
```

Comment:
Thinking about naming here. `getAdjustedDate` is a bit opaque and tough to make a guess at its function without inspecting the code (especially when being exported for consumption by other files).

`normalizeDate` -- still kind of opaque, but suggests some sort of transformation to get to a standard.
`adjustTimezoneOffset`
`applyTimezoneOffset`
`valueToLocalDate`
`normalizeForTimezone`